x-pack/elastic-agent/pkg/composable/controller.go

```diff
@@ -18,9 +18,14 @@ import (
 	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/errors"
 	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/transpiler"
 	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/config"
+	corecomp "github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/core/composable"
 	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/core/logger"
 )
 
+var FetchContextProviders = map[string]string{
+	"kubernetes_secrets": "",
+}
```

Comment:
This really removes the plugin coding pattern. Why not just add a `FetchContextProvider` and allow it to register just like the other providers? This is not really following that same pattern.

Reply:
Changed it based on https://github.com/elastic/beats/pull/24789#discussion_r603410573